### PR TITLE
gocli: add an option to enable FIPS

### PR DESCRIPTION
**Pull request by jean-edouard**

### CODE REVIEW

**Review by dhiller — COMMENTED** [review]
/approve

/cc @brianmcarey could you take a look please?

File: cluster-provision/gocli/cmd/run.go [hunk 1]
```diff
@@ -512,22 +523,29 @@ func run(cmd *cobra.Command, args []string) (retErr error) {
...
+		err = waitForVMToBeUp(prefix, nodeName)
+		if err != nil {
+			return err
 		}
 
-		if err != nil {
-			return fmt.Errorf("could not establish a connection to the node after a generous timeout: %v", err)
+		if fipsEnabled {
+			success, err := docker.Exec(cli, nodeContainer(prefix, nodeName), []string{
+				"/bin/bash", "-c", "ssh.sh sudo fips-mode-setup --enable && ( ssh.sh sudo reboot || true )",
```
Comment (xpivarc) [inline cluster-provision/gocli/cmd/run.go:533]: Maybe split this into two to be able to distinguish the failure.

File: cluster-provision/gocli/cmd/run.go [hunk 2]
```diff
@@ -461,18 +467,23 @@ func run(cmd *cobra.Command, args []string) (retErr error) {
 		}
 
 		if hugepages2Mcount > 0 {
-			args := fmt.Sprintf("hugepagesz=2M hugepages=%d", hugepages2Mcount)
-			if len(kernelArgs) > 0 {
-				args = fmt.Sprintf("%s %s", args, kernelArgs)
-			}
-			additionalArgs = append(additionalArgs, "--additional-kernel-args", shellescape.Quote(args))
+			kernelArgs += fmt.Sprintf(" hugepagesz=2M hugepages=%d", hugepages2Mcount)
```
Comment (xpivarc) [inline cluster-provision/gocli/cmd/run.go:470]: shame we don't represent `kernelArgs` as `[]string`

**Comment by dhiller** [comment]
/override check-up-kind-1.23-vgpu

@brianmcarey this one is very flaky, I'd rather make it optional. WDYT?

**Comment by brianmcarey** [comment]
> /override check-up-kind-1.23-vgpu
> 
> @brianmcarey this one is very flaky, I'd rather make it optional. WDYT?

I opened an issue for it here - https://github.com/kubevirt/kubevirtci/issues/925 - it makes sense to make optional for now as it is not useful at the moment

**Pull request merged**